Autoescuela/Autoescuela/Coordinators/AppCoordinator/AppCoordinator.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  AppCoordinator.swift
+//  Autoescuela
+//
+//  Created by user159106 on 6/6/20.
+//  Copyright © 2020 user159106. All rights reserved.
+//
+
+import UIKit
+
+public class AppCoordinator: Coordinator {
+    
+    public enum AppCoordinatorState {
+        case initial
+        case willShowLoginFlow
+        case willShowMainMenuFlow
+        case didShowLoginFlow
+        case didShowMainMenuFlow
+    }
+    
+    let navigationController: UINavigationController
+    var currentState: AppCoordinatorState
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+        self.currentState = .initial
+        
+    }
+    
+    public func start() {
+        currentState = .initial
+        loop()
+    }
+    
+    func loop() {
+        currentState = next(currentState)
+        switch currentState {
+        case .willShowLoginFlow:
+            goToLoginFlow()
+        case .willShowMainMenuFlow:
+            goToMainMenuFlow()
+        case .didShowMainMenuFlow , .didShowLoginFlow , .initial :
+            fatalError("Unexpecrted cases")
+        }
+    }
+    
+    func next(_ state: AppCoordinatorState) -> AppCoordinatorState {
+        switch state {
+        case .initial:
+            return .willShowLoginFlow
+        case .didShowLoginFlow:
+            return .willShowMainMenuFlow
+        case .didShowMainMenuFlow:
+            return .willShowLoginFlow
+        case .willShowLoginFlow ,.willShowMainMenuFlow:
+            return state
+        }
+    }
+    
+    func goToMainMenuFlow() {
+        
+    }
+    
```

Comment:
This funcion is empty if you ain't need it you could remove it

Reply:
I prefer to leave it because this way the loop functions definition its completed. But maybe a comment explaining why is empty would be good

Reply:
Your’re right this should contain the same code as the scene function in the sceneDelegate because its the entry point for decices with an ios version lower than 13

Reply:
Yeah, that's basically the point, we are just looking a function without code, so at first i will think that it should be deleted.